src/main/kotlin/net/axay/hglaborlobby/gui/guis/PetGUI.kt

```diff
@@ -0,0 +1,139 @@
+package net.axay.hglaborlobby.gui.guis
+
+import net.axay.kspigot.chat.KColors
+import net.minecraft.server.v1_16_R3.EntityInsentient
+import net.minecraft.server.v1_16_R3.PathfinderGoalSelector
+import org.bukkit.DyeColor
+import org.bukkit.Material
+import org.bukkit.craftbukkit.v1_16_R3.entity.CraftEntity
+import org.bukkit.entity.*
+import net.axay.hglaborlobby.pathfinding.LaborPathfinderMoveToPlayer
+import net.axay.kspigot.extensions.bukkit.*
+import net.axay.kspigot.gui.*
+import net.axay.kspigot.items.*
+import net.axay.kspigot.runnables.task
+import net.minecraft.server.v1_16_R3.PathfinderGoalFloat
+import org.bukkit.Sound
+import org.bukkit.inventory.EquipmentSlot
+import org.bukkit.inventory.ItemStack
+
+
+object PetGUI {
+
+    data class Pet(val name: String, val entity: EntityType, val icon: Material) {
+
+        lateinit var petEntity: Entity
+
+        fun despawn() {
+            petEntity.remove()
+        }
+
+        fun spawn(owner: Player) {
+            petEntity = owner.world.spawnEntity(owner.location, entity)
```

Comment:
please do
```
val currentPetEntity = owner.world.spawnEntity(owner.location, entity)
petEntity = currentPetEntity
// and now continue currentPetEntity in the spawn function, do not use the lateinit var petEntity down here
```
this way you won't need the casts anymore, and overall this is more safe

Reply:
like now?

Reply:
yes